### PR TITLE
[Empathetic Dialogues] Switch to DialogTeacher

### PR DESCRIPTION
**Pull request by EricMichaelSmith**
# Patch description
- Switch EmpatheticDialogues over to `DialogTeacher` to allow for efficient eval of this dataset in distributed mode
- Delete an unused teacher and unused flags and `Message` keys
- Also switch `EDPersonaTopicifierTeacher`, a version of EmpatheticDialogues that includes personas, over to `DialogTeacher`

# Testing steps
- Checked that the following pass:
  - `pytest tests/tasks/test_empathetic_dialogues.py`
  - `pytest parlai/tasks/empathetic_dialogues/test.py`
  - `pytest parlai/tasks/blended_skill_talk/test.py`
  - `pytest tests/tasks/test_style_gen_teachers.py`
- Compared the output of `parlai dd -t empathetic_dialogues -v` and `parlai dd -t parlai.tasks.blended_skill_talk.agents:EDPersonaTopicifierTeacher` before and after this PR

### CODE REVIEW

File: parlai/tasks/empathetic_dialogues/agents.py [hunk 1]
```diff
@@ -4,57 +4,37 @@
...
 DEFAULT_TRAIN_EXPERIENCER_ONLY = False
-DEFAULT_REMOVE_POLITICAL_CONVOS = False
 
 
-class EmpatheticDialoguesTeacher(FixedDialogTeacher):
+class EmpatheticDialoguesTeacher(DialogTeacher):
     def __init__(self, opt, shared=None):
-        super().__init__(opt, shared)
-        self.opt = opt
+        self.datatype = opt['datatype']
```
Comment (stephenroller) [inline parlai/tasks/empathetic_dialogues/agents.py:23]:
```suggestion
```

this is a dangerous attribute to set yourself. the parent class sets it.

File: parlai/tasks/empathetic_dialogues/agents.py [hunk 2]
```diff
@@ -71,40 +51,18 @@ def add_cmdline_args(
...
-        if self.opt.get('fasttextloc') is not None and self.opt.get('prepend', -1) > 0:
-            try:
-                import fastText
-            except ImportError:
-                raise ImportError("Please run 'pip install fasttext'.")
-            ftpath = self.opt['fasttextloc']
-            ftmodel = fastText.FastText.load_model(ftpath)
-
-        with PathManager.open(self.datapath) as f:
+        print('loading: ' + path)
```
Comment (stephenroller) [inline parlai/tasks/empathetic_dialogues/agents.py:58]:
```suggestion
        logging.debug('loading: ' + path)
```

File: parlai/tasks/empathetic_dialogues/agents.py [hunk 3]
```diff
@@ -181,98 +117,40 @@ def _setup_data(self, base_datatype):
...
                 experiencer_text_dialogue = []
                 responder_text_dialogue = []
 
         # Add in the final episode
-        self.data += self._select_dialogues_to_add(
+        data += self._select_dialogues_to_add(
             experiencer_text_dialogue, responder_text_dialogue
         )
 
+        for episode in data:
```
Comment (stephenroller) [inline parlai/tasks/empathetic_dialogues/agents.py:131]: in an ideal world we would make this entire function an iterator that didn't explicitly load data into memory, which speeds up `parlai dd` and lowers peak memory usage.

it's fine for this PR

**Review by stephenroller — APPROVED** [review]
very nice! great cleanup and faster to boot!

**Comment by stephenroller** [comment]
Make sure you merge in master to get the fixes to CI, and check `pytest parlai/tasks/empathetic_dialogues/test.py` (which you seemed to skip since it has two sets of tests)

**Comment by EricMichaelSmith** [comment]
> Make sure you merge in master to get the fixes to CI, and check `pytest parlai/tasks/empathetic_dialogues/test.py` (which you seemed to skip since it has two sets of tests)

Good point, checked

File: parlai/tasks/empathetic_dialogues/agents.py [hunk 4]
```diff
@@ -4,57 +4,28 @@
...
+from parlai.core.opt import Opt
+from parlai.core.teachers import DialogTeacher
 from .build import build
 
 
 DEFAULT_TRAIN_EXPERIENCER_ONLY = False
-DEFAULT_REMOVE_POLITICAL_CONVOS = False
 
 
-class EmpatheticDialoguesTeacher(FixedDialogTeacher):
```
Comment (tempdeltavalue) [inline parlai/tasks/empathetic_dialogues/agents.py:25]: ImportError: cannot import name 'EmpatheticDialoguesTeacher' from 'parlai.core.teachers'

Comment (EricMichaelSmith) [reply parlai/tasks/empathetic_dialogues/agents.py:25]: Can you paste the command that gives that error?

**Pull request merged**